doozer/doozerlib/distgit.py

```diff
@@ -1558,11 +1558,11 @@ def _mapped_image_from_stream(self, image, original_parent, dfp):
             out, _ = exectools.cmd_assert(cmd, retries=3)
 
             # It does. Use this to rebase FROM directive
-            digest = json.loads(out)['digest']
+            mapped_image_data = json.loads(out)
 
-            mapped_image = f'{labels["name"]}@{digest}'
             # if upstream equivalent does not match ART's config, add a warning to the Dockerfile
-            if mapped_image != stream.image:
+            mapped_image = f'{labels["name"]}@{mapped_image_data["digest"]}'
+            if mapped_image_data['name'].split(':')[-1] != stream.image.split(":")[-1]:
```

Comment:
This looks like a tricky piece of code, can we have 1-2 unit tests so we can be sure of its correctness?

Reply:
And should we tweak the log message on 1571 to indicate the values we are comparing?